src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -1149,6 +1149,25 @@ private static void PartialMethodChecks(SourceOrdinaryMethodSymbol definition, S
         {
             Debug.Assert(!ReferenceEquals(definition, implementation));
 
+            MethodSymbol constructedDefinition = definition;
+            if (implementation.IsGenericMethod)
+            {
+                constructedDefinition = definition.Construct(implementation.TypeArgumentsWithAnnotations);
+            }
+
+            bool returnTypesEqual = constructedDefinition.ReturnTypeWithAnnotations.Equals(implementation.ReturnTypeWithAnnotations, TypeCompareKind.AllIgnoreOptions);
+            if (!returnTypesEqual
+                && !SourceMemberContainerTypeSymbol.IsOrContainsErrorType(implementation.ReturnType)
+                && !SourceMemberContainerTypeSymbol.IsOrContainsErrorType(definition.ReturnType))
```

Comment:
If the types are distinct, why not report that, regardless of whether the types contain errors?

Reply:
This is based on patterns I saw in `CheckOverrideMember`, it seemed like reducing the cascading errors was best.

Reply:
> it seemed like reducing the cascading errors was best.

Ok, but it doesn't seem like a cascading error to me though because the types are different.

---
In reply to: [442526990](https://github.com/dotnet/roslyn/pull/45128#discussion_r442526990) [](ancestors = 442526990)